### DISCUSSION TITLE
Naming slides (stable anchor links)

### DISCUSSION THREAD

**Discussion by eyssette** [marp-team/marp#161] [Ideas] [answered]
Hello!
If i want to link to a slide, i have to use its ID. But if I change the document, i have to change the links to the new IDs.
It would be great to have the possibility to name a slide, so that anchor links would be stable.
Is it already possible? I doubt so, but maybe someone has an idea !
Thank you!

**Comment by yhatt** [comment]
[A traditional anchor link to `id`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#linking_to_an_element_on_the_same_page) is already working correctly on the exported PDF (require to enable HTML). 

```markdown
# Page 1

[Jump to third page](#third-page)

---

...

---

### Page 3 <a id="third-page"></a>
```

See [naming-slides.pdf](https://github.com/marp-team/marp/files/7128514/naming-slides.pdf).

**Reply by dontlookherepls** [reply]
not working on marp-cli, it always brings me back to the firts page.

**Reply by yhatt** [reply]
@LucaErr Markdown must convert into PDF, not other formats such as HTML and PPTX. In addition, you have to enable HTML through `--html` to enable named anchor.

**Reply by dontlookherepls** [reply]
so there isn't possibility to use it with animations... am i right ?

**Reply by yhatt** [reply]
Right. _Marp is historically made for static PDF slides_, and Marp team treats animations and interactivities in HTML as a bonus. Thus, we have a low priority for better named anchor support in HTML.

**Reply by amitbha** [reply]
Plain text is the current trend. Hope someday the HTML is treated as first class.

**Comment by captainabap** [comment]
Stable Anchor Links would be a great improvement. I use Marp for training slides. They improve every day, so the slide number is changing often. A stable link would be very helpful. 

Exporting to PDF is not an option, because the slides are presented as HTML.

**Reply by yhatt** [reply]
We are considering to implement header rendering with auto-generated named anchor, for make compatible with common Markdown renderers (https://github.com/marp-team/marp-core/issues/299, #377). This should become a strong motivation to support custom named anchor in HTML exported with [`bespoke` template](https://github.com/marp-team/marp-cli#bespoke-template-default).

**Comment by yhatt** [comment, marked as answer]
**UPDATE**: The latest [Marp CLI v2.5.0](https://github.com/marp-team/marp-cli/releases/tag/v2.5.0) and [Marp for VS Code v2.6.0](https://github.com/marp-team/marp-vscode/releases/tag/v2.6.0) have added support for links to the named anchor.

This updates are including [assigning auto-generated slugs to Markdown headings](https://github.com/marp-team/marp-core/pull/338) and [the support of anchor links in the generated HTML from Marp Markdown](https://github.com/marp-team/marp-cli/pull/518). Links to the Markdown heading `[link](#heading)`, that is suggestable by VS Code (Markdown Language Server), would have to work correctly.

### Slug (Link to Markdown heading)

```markdown
# Title

[Jump to Hello](#hello)

---

## Contents

- ...
- ...

---

## Hello

Example
```

In general Markdown document, a link of "Jump to Hello" can actually jump to "Hello" heading. And now, Marp can jump to the slide page 3 too.

### Named anchor (with raw HTML)

```markdown
[Jump to closing slide](#closing-slide)

---

...

---

<a id="closing-slide"></a>

Thanks!
```

Marp also supports the anchor defined through `<a id="..."></a>` or `<a name="..."></a>`. It's useful when there is not any Markdown headings in the slide.

To use `<a>` HTML tag, please don't forget to enable raw HTML tags in the Marp tool you are using. (`--html` option in Marp CLI, and `markdown.marp.enableHtml` preference in Marp for VS Code)
